resize-observer/devicepixel2.html

```diff
@@ -0,0 +1,152 @@
+<!doctype html>
+<link rel="match" href="devicepixel2-ref.html">
+<meta name="assert" content="Device pixel content box sizes and pixel snapping are correct in Resize Observer callback">
+<style>
+  html, body {
+    margin: 0;
+  }
+  .outer {
+    width: 100px;
+    display: flex;
+  }
+  .outer>* {
+    flex: 1;
+    height: 20px;
+  }
+</style>
+<body>
+  <div></div>
+</body>
+<script>
+const patternSize = 4;
+
+function createPatternDataURL() {
+  const ctx = document.createElement('canvas').getContext('2d');
+  ctx.canvas.width = patternSize;
+  ctx.canvas.height = patternSize;
+
+  const b = [0, 0, 0, 255];
+  const t = [0, 0, 0, 0];
+  const r = [255, 0, 0, 255];
+  const g = [0, 255, 0, 255];
+
+  const imageData = new ImageData(patternSize, patternSize);
+  imageData.data.set([
+    b, t, t, r,
+    t, b, g, t,
+    t, r, b, t,
+    g, t, t, b,
+  ].flat());
+  ctx.putImageData(imageData, 0, 0);
+  return ctx.canvas.toDataURL();
+}
+
+const patternURL = createPatternDataURL();
+
+/**
+ * Set the pattern's size on this element so that it draws where
+ * 1 pixel in the pattern maps to 1 devicePixel and then set
+ * its position so it's aligned to the pattern of the body element.
+ */
+function setPattern(elem, x, y) {
+  const oneDevicePixel = 1 / devicePixelRatio;
+  const patternPixels = oneDevicePixel * patternSize;
+  elem.style.backgroundImage = `url("${patternURL}")`;
+  elem.style.backgroundSize = `${patternPixels}px ${patternPixels}px`;
+  elem.style.backgroundPosition = `${-x * oneDevicePixel}px ${-y * oneDevicePixel}px`;
+}
+
+/*
+This test creates elements like this
+
+  <body>
+    <div>
+      <div class="outer"><div></div><div></div></div>
+      <div class="outer"><div></div><div></div><div></div></div>
+      <div class="outer"><div></div><div></div><div></div><div></div></div>
+      ...
+    </div>
+  </body>
+
+It has the first `<div>` starting at the top left corner of the page, the
+same as the body so they should both have a devicePixel position of 0,0
+
+It then sets a repeating background-image and background-size on the
+body element and all of the .outer children elements so they have a 4x4
+checkerboard image at a size that should draw each pixel of the 4x4
+checkerboard pattern in single devicePixels.
+
+The devicePixelContentBoxSize of all the elements is queried with
+a ResizeObserver.
+
+Finally the background-position of each child element is set so that
+the patterns should align.
```

Comment:
This chunk of the comment doesn't match what the test actually does.

In particular, three things to address:
 - The phrase ".outer children" is ambiguous & sounds like it's referring to the .outer elements themselves (it sounds like "children who have class .outer"); and that's confusing because the .outer elements don't get any background at all.  Maybe just rephrase as "inner children"? (since you're talking about the innermost things)

 - The current wording suggests that we set the background-position in a final pass (with the image + size having been set earlier).  But in fact, we set the pattern/size/position all at once.

 - The current wording suggests that we set the body's background first and then handle all of the elements later.  But in fact, we handle the body *last*.

Could you update the wording here to address these?  (In the case of the last item, we could just move the `setPattern(document.body, 0, 0);` to be the first thing that happens in the ResizeObserver callback. It is useful to keep it in the callback, to ensure that that the test will fail if the callback never fires.)